R/handler_validations.R

```diff
@@ -0,0 +1,152 @@
+#' @title Check Correctness of `data_set`
+#'
+#' @description This auxiliary function checks that the type of the object
+#' provided in `data_set` is a `data.frame` and that the specified columns
+#' are included in it.
+#'
+#' @param data_set `data.frame`.
+#' @param columns A vector of column names that should be present in
+#' `data_set`.
+#'
+#' @return This function does not return a value.
+#'
+#' @keywords internal
+
+check_dataset <- function(data_set, columns) {
+  # check for data.frame type
+  checkmate::assert_data_frame(
+    data_set,
+    min.rows = 1L
+  )
+
+  # check for columns included
+  checkmate::assert_names(
+    names(data_set),
+    must.include = columns
+  )
+}
+
+#' @title Check Correctness of Inputs in `make_vaccineff_data`
+#'
+#' @description This auxiliary function checks the correctness of the inputs
+#' provided to `make_vaccineff_data`.
+#'
+#' @inheritParams make_vaccineff_data
+#' @return This function does not return a value.
+#'
+#' @keywords internal
+
+check_vaccineff_inputs <- function(data_set,
+                                   outcome_date_col,
+                                   censoring_date_col,
+                                   vacc_date_col,
+                                   vacc_name_col,
+                                   vaccinated_status,
+                                   unvaccinated_status,
+                                   immunization_delay,
+                                   end_cohort,
+                                   match,
+                                   exact,
+                                   nearest,
+                                   take_first) {
+  # Check names of variables
+  checkmate::assert_string(outcome_date_col)
+  checkmate::assert_string(censoring_date_col, null.ok = TRUE)
+  checkmate::assert_character(vacc_date_col, min.len = 1L)
+  checkmate::assert_character(
+    vacc_name_col,
+    min.len = length(vacc_date_col),
+    null.ok = TRUE
+  )
+
+  # Check data.frame
+  check_dataset(data_set = data_set,
+    columns = c(outcome_date_col, censoring_date_col,
+                vacc_date_col, vacc_name_col)
+  )
+
+  # Check variable types
+  checkmate::assert_date(
+    data_set[[outcome_date_col]]
+  )
+
+  for (i in seq_along(vacc_date_col)) { #loop along vacc_date_col (can be many)
+    checkmate::assert_date(
+      data_set[[vacc_date_col[i]]]
+    )
+  }
```

Comment:
The examples are taking too long because of the size of the example dataset. We're currently working on this in issue https://github.com/epiverse-trace/vaccineff/issues/82. However, we will take a look at this suggestion in issue https://github.com/epiverse-trace/vaccineff/issues/81as well.